02.wo_Defects/st_underrun.c

```diff
@@ -50,7 +50,7 @@ void st_underrun_002_func_001 (st_underrun_002_s_001 s)
 	 int len = strlen(s.buf) - 1;
 	 for (;s.buf[len] != 'Z';len--)  /*Tool should not detect this line as error*/ /* No Stack Under RUN error */
 	 {
-	    if ( len < 0 )
+	    if ( len <= 0 )
```

Comment:
if we use the original condition, then following the loop iteration where len == 0, we will retry the loop condition and end up reading s.buf[-1].